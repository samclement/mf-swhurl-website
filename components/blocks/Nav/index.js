import React from 'react'
import Link from '../../elements/Link'

export default () => {
  return (
    <nav>
      <Link href="/" label="Home" />
      &nbsp;|&nbsp;
      <Link href="/about" label="About" />
      &nbsp;|&nbsp;
      <Link href="/contact" label="Contact" />
    </nav>
  )
}
