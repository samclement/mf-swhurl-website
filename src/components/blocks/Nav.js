import React from 'react'
import NavLink from '../elements/NavLink'
import styled from 'styled-components'

const StyledTitle = styled.div`
  font-weight: bold;
  text-transform: capitalize;
`

const StyledNav = styled.nav`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
`

export default ({ items, gender }) => {
  return (
    <StyledNav>
      <StyledTitle>{gender}</StyledTitle>
      {items.map((item, i) => {
        const url = `/plp?url=${item.url}`
        return <NavLink key={i} href={url} as={item.url} label={item.name} />
      })}
    </StyledNav>
  )
}
