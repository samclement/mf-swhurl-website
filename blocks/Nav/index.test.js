/* eslint-env jest */

import React from 'react'
import { render, shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Nav from './index.js'

describe('With Snapshot Testing', () => {
  it('Nav has 3 sections', () => {
      const wrapper = render(<Nav/>)
      expect(wrapper).toMatchSnapshot()
    })
})

