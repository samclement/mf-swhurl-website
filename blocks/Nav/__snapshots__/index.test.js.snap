// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Nav has 3 sections 1`] = `
<nav>
  <a
    class="sc-bdVaJa kPUKaM"
    href="/"
  >
    Home
  </a>
   | 
  <a
    class="sc-bdVaJa kPUKaM"
    href="/about"
  >
    About
  </a>
   | 
  <a
    class="sc-bdVaJa kPUKaM"
    href="/contact"
  >
    Contact
  </a>
</nav>
`;
